imap/mailbox.c

```diff
@@ -4081,14 +4084,31 @@ static int mailbox_update_sieve(struct mailbox *mailbox,
         char *errors = NULL;
 
         r = sievedir_put_script(mailbox->sievedir, name, content, &errors);
+        if (r) {
+            if (SIEVEDIR_INVALID) {
```

Comment:
I think you mean to be comparing r to SIEVEDIR_INVALID here?

Reply:
Yep.  Fixed